src/Swashbuckle.SwaggerGen/Annotations/XmlCommentsModelFilter.cs

```diff
@@ -34,7 +34,11 @@ public void Apply(Schema model, ModelFilterContext context)
                 var jsonProperty = context.JsonObjectContract.Properties[entry.Key];
                 if (jsonProperty == null) continue;
 
-                ApplyPropertyComments(entry.Value, jsonProperty.PropertyInfo());
+                var propertyInfo = context.JsonObjectContract.UnderlyingType.GetProperty(jsonProperty.UnderlyingName);
```

Comment:
At first glance, it seems the above line could be reduced to:

    var propertyInfo = jsonProperty.PropertyInfo();

Maybe I'm missing something and if so, perhaps a unit test would help make the intention here a  little clearer